R/deploySite.R

```diff
@@ -35,61 +38,47 @@ deploySite <- function(siteDir = getwd(),
                        recordDir = NULL,
                        ...) {
 
-  check_installed(
-    "rmarkdown",
-    version = "0.9.5.3",
-    reason = "to deploy websites"
-  )
-
-  # validate and normalize siteDir
   check_directory(siteDir)
-  siteDir <- normalizePath(siteDir)
+  isQuarto <- file.exists(file.path(siteDir, "_quarto.yml")) ||
+    file.exists(file.path(siteDir, "_quarto.yaml"))
 
-  # switch to siteDir for duration of this function
-  oldwd <- setwd(siteDir)
-  on.exit(setwd(oldwd), add = TRUE)
-
-  # discover the site generator
-  siteGenerator <- rmarkdown::site_generator(siteDir)
-  if (is.null(siteGenerator)) {
-    stop("index file with site entry not found in ", siteDir)
+  quiet <- identical(match.arg(logLevel), "quiet")
+  if (isQuarto) {
+    site <- quartoSite(siteDir, quiet = quiet)
+  } else {
+    site <- rmarkdownSite(siteDir, quiet = quiet)
   }
 
   # render locally if requested
-  render <- match.arg(render)
+  render <- arg_match(render)
```

Comment:
note: Client-side rendering by rsconnect is supported by `deploySite` but not `deployDoc`. Rendering is performed external to rsconnect for single file deploys.

Reply:
If I have time, I think I should implement this for `deployDoc()` too, to bring closer to `quarto_publish_doc()`.